get_gcs.go

```diff
@@ -127,11 +129,22 @@ func (g *GCSGetter) GetFile(dst string, u *url.URL) error {
 	if err != nil {
 		return err
 	}
-	return g.getObject(ctx, client, dst, bucket, object)
+	return g.getObject(ctx, client, dst, bucket, object, fragment)
 }
 
-func (g *GCSGetter) getObject(ctx context.Context, client *storage.Client, dst, bucket, object string) error {
-	rc, err := client.Bucket(bucket).Object(object).NewReader(ctx)
+func (g *GCSGetter) getObject(ctx context.Context, client *storage.Client, dst, bucket, object, fragment string) error {
+	var rc *storage.Reader
+	var err error
+	fragmentHasGeneration := regexp.MustCompile("^\\d+$").MatchString(fragment)
```

Comment:
Is the fragment able to ever have anything _other_ than generation/version information?  (I know nothing about GCS objects). If so, is this information ever useful?

Reply:
I'll take the `if not` case :)

If the fragment can only ever be a generation number, can we skip the regex and go straight to ParseInt?

```go
if len(fragment) > 0 {
  // has generation code here
} else {
  // no generation code here
}
```

My concern is that people will intend to get a specific generation with a fragment like: `#v36` but instead of an error due to the `v`, they will silently be given _another_ generation than they expected.

Reply:
Oh great point. Seconded.

Reply:
not that I know of !

Reply:
Well I suppose the reason I have this check here is if the user puts junk in the hash it will prevent the parsing the of the hash as an int64 from erroring out. It's sort of a quality of life thing. If the hash is junk, just pull gcs object remotely without considering the generation. Thats why I decided to include the check. What do you think?

Reply:
Thats a good point about that giving an error if the generation is given in another format ! I didn't think about that. I agree with your perspective!